model/task/task.go

```diff
@@ -2557,35 +2578,67 @@ func ArchiveMany(tasks []Task) error {
 	defer session.EndSession(ctx)
 
 	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
-		oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
-		_, err = oldTaskColl.InsertMany(ctx, archived)
-		if err != nil {
-			return nil, err
+		var err error
+		if len(toArchive) > 0 {
+			oldTaskColl := evergreen.GetEnvironment().DB().Collection(OldCollection)
+			_, err = oldTaskColl.InsertMany(sessCtx, toArchive)
+			if err != nil {
+				return nil, errors.Wrap(err, "archiving tasks")
+			}
 		}
-
-		taskColl := evergreen.GetEnvironment().DB().Collection(Collection)
-		_, err = taskColl.UpdateMany(ctx, bson.M{
-			IdKey: bson.M{
-				"$in": taskIds,
-			},
-		},
-			bson.M{
-				"$unset": bson.M{
-					AbortedKey:   "",
-					AbortInfoKey: "",
-				},
-				"$inc": bson.M{
-					ExecutionKey: 1,
+		if len(tasksIds) > 0 {
+			// TODO (EVG-17322): Remove RunCommand call and replace with aggregate + merge
+			result := evergreen.GetEnvironment().DB().RunCommand(sessCtx,
```

Comment:
I discussed with Bynn but I'm concerned about introducing this because A) it introduces tech debt since it’s a bit harder to read and is very specific, and B) it’s likely much more expensive than what we have now, which is already expensive.

I'm not convinced we need to do any of the backwards compatible calls since this change will be baked in for a few months before users can start. Really what we need to do is:
1) unset anything that needs to be unset
2) Update execution to execution+1 for non-exec tasks, and LPE+1 for exec tasks (could probably use aggregation stage $project or something), 
3) Update LPE to be LPE +1 for exec tasks.

Seems like we could have 1 call for non-exec and 1 call for exec, or do a $split aggregation stage. It's not clear to me from your ticket what $merge is needed for but bynn  and/or i will sync with you.